### ISSUE TITLE
Upgrade of expedient from v0.4.3 to v0.5 fails

### ISSUE THREAD

**Issue by vkotronis** [fp7-ofelia/ocf#150]
The upgrade of Expedient to the new version of the OCF (v0.5) fails with the following error due to the manage.py file:

Traceback (most recent call last):
  File "manage.py", line 13, in <module>
    main()
  File "manage.py", line 10, in main
    execute_manager(settings)
  File "/usr/lib/pymodules/python2.6/django/core/management/**init**.py", line 438, in execute_manager
    utility.execute()
  File "/usr/lib/pymodules/python2.6/django/core/management/**init**.py", line 379, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/lib/pymodules/python2.6/django/core/management/base.py", line 191, in run_from_argv
    self.execute(_args, *_options.**dict**)
  File "/usr/lib/pymodules/python2.6/django/core/management/base.py", line 219, in execute
    self.validate()
  File "/usr/lib/pymodules/python2.6/django/core/management/base.py", line 249, in validate
    num_errors = get_validation_errors(s, app)
  File "/usr/lib/pymodules/python2.6/django/core/management/validation.py", line 28, in get_validation_errors
    for (app_name, error) in get_app_errors().items():
  File "/usr/lib/pymodules/python2.6/django/db/models/loading.py", line 146, in get_app_errors
    self._populate()
  File "/usr/lib/pymodules/python2.6/django/db/models/loading.py", line 61, in _populate
    self.load_app(app_name, True)
  File "/usr/lib/pymodules/python2.6/django/db/models/loading.py", line 76, in load_app
    app_module = import_module(app_name)
  File "/usr/lib/pymodules/python2.6/django/utils/importlib.py", line 35, in import_module
    __import__(name)
  File "../../vt_plugin/communication/**init**.py", line 1, in <module>

  File "../../vt_plugin/communication/southCommInterface.py", line 4, in <module>
  File "../../vt_plugin/controller/dispatchers/ProvisioningResponseDispatcher.py", line 11, in <module>
  File "../../vt_plugin/controller/vtAggregateController/vtAggregateController.py", line 10, in <module>
  File "../../vt_plugin/controller/vtAggregateController/forms/forms.py", line 13, in <module>
  File "/usr/lib/pymodules/python2.6/django/forms/models.py", line 226, in **new**
    opts.exclude, opts.widgets, formfield_callback)
  File "/usr/lib/pymodules/python2.6/django/forms/models.py", line 166, in fields_for_model
    opts = model._meta
AttributeError: 'module' object has no attribute '_meta'
FATAL ERROR: Could not synchronize database using manage.py...

 info#  Using default rollback module...
It seems something went wrong.. Now you will be able to rollback to the previous state of the filesystem and database. Alternatively you can try to solve the problem manually, avoiding the rollback mechanism. Do you want to rollback now?. Do you confirm (Y/N)?

After that the island manager can of course roll back, but the upgrade fails to install the new version. I have not checked the other components, but I assume the behavior is similar because this is a django error that affects all the components.

**Comment by CarolinaFernandez** [comment]
This specific error is due to an error in the `pre-upgrade-hook` script where the old plugin folders are being moved to another location. Since that is not being done, the old *.pyc files (within `openflow` and `vt_plugin`) are interfering with the new ones (in the `plugins` folder).

To solve this, the `pre-upgrade-hook` needs to be updated with the proper (relative) path to these folders (or alternatively delete these folders manually, in order to keep on testing)

But this is not the real deal: the subjacent error is due to some other condition, that may be one of this:
- Private and public repo have tracked different file modes and is seen as a difference when pulling from public repo
- Some templates that existed in `expedient/src/python/expedient/ui/html` were moved to `expedient_geni/views/templates`, but this was not committed and turns out as a difference when pulling from public repo. When these changes are committed, `ofelia.stable` points to commit `023e74315e` that is already on Github and can pull from it without any apparent problem
- ...?

Some tests need to be made to determine if the problem is due to permissions, real conflicts on files or any other stuff related to GIT commits history.

**Comment by CarolinaFernandez** [comment]
Fixed in the v0.5 release by adding the following code to the `pre-upgrade-hook`:

```
# Moving basic plugins from old directory to new location
# to avoid loading their *.pyc files
if [ -d $OFELIA_DIR/expedient/src/python/vt_plugin ]; then
    rm -r $OFELIA_DIR/expedient/src/python/vt_plugin || error "Could not remove 
    old plugin 'vt_plugin'"
fi

if [ -d $OFELIA_DIR/expedient/src/python/openflow ]; then
    rm -r $OFELIA_DIR/expedient/src/python/openflow || error "Could not remove 
    old plugin 'openflow'"
fi
```

From now on this folder should not exist anymore and therefore should not need to remove any *.pyc files here.
